File: reduce.py
words = {}
with open("words.lst", "w") as fout:
    count = 0
    with open("all.lst") as fin:
        for line in fin:
            count += 1
            if count > 1000:
                break
            word = line.split(",", 1)[0]
            words[word] = True
            fout.write(word + "\n")

with open("cc.zh.300-small.vec", "w") as fout:
    fout.write(str(len(words)) + " 300\n")
    with open("cc.zh.300.vec") as fin:
        for line in fin:
            word = line.split(" ", 1)[0]
            if word in words:
                fout.write(line)

File: reduce.js
import { TextLineStream } from "@std/streams";

async function loadDict(threshold) {
  const dict = [];
  const file = await Deno.open("jieba/extra_dict/dict.txt.big");
  const lineStream = file.readable
    .pipeThrough(new TextDecoderStream())
    .pipeThrough(new TextLineStream());
  for await (const line of lineStream) {
    const arr = line.split(" ");
    const word = arr[0];
    const count = parseInt(arr[1]);
    if (count > threshold) {
      dict.push([word, count]);
    }
  }
  dict.sort((a, b) => b[1] - a[1]);
  return dict;
}

const threshold = 99;
const dict = await loadDict(threshold);
const output = dict.map(x => x.join(",")).join("\n");
Deno.writeTextFileSync("all.lst", output);
